invoices/update.go

```diff
@@ -125,6 +125,21 @@ func updateMpp(ctx *invoiceUpdateCtx,
 	inv *channeldb.Invoice) (*channeldb.InvoiceUpdateDesc,
 	HtlcResolution, error) {
 
+	// Reject HTLCs to AMP invoices if they are missing an AMP payload, and
```

Comment:
In practice this would only happen due to user error at the RPC level when crafting their send request right?

Reply:
it could happen for any number of reasons, even just differences/incorrect handling of AMP invoices across implementations. this check is necessary though because without it you can cause the registry to panic, as the logic below is gated purely on the presence of the amp record.